platform/spire/src/template.py

```diff
@@ -40,3 +41,42 @@ def template(filename_or_contents, keys, load=True) -> str:
 def template_all(filename_or_contents, keys_iterable, load=True) -> list:
     templ = Template(filename_or_contents, load=load)
     return [templ.template(keys) for keys in keys_iterable]
+
+
+def string_substitute(text: str, keys: dict) -> str:
```

Comment:
Suggestion: just use ordinary python format string syntax; this function can be defined as:
```
def string_substitute(text, keys):
    return text.format(**keys)
```

Reply:
Yeah, I'm really not sure why I didn't just go with that.